### PR TITLE
Add helper for mocking FLAGS across all modules.

### PR DESCRIPTION
**Pull request by skschneider**
No need to patch module1.FLAGS and module2.FLAGS separately.

### CODE REVIEW

**Comment by voellm** [comment]
@ivansmf you want to take this review?  You love mocks.

On Wed, Nov 25, 2015 at 9:26 PM, skschneider <notifications@github.com>
wrote:

> No need to patch module1.FLAGS and module2.FLAGS separately.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/678
> Commit Summary
>
>    - Add helper for mocking FLAGS across all modules.
>
> File Changes
>
>    - *A* tests/__init__.py
>    <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/678/files#diff-0>
>    (13)
>    - *A* tests/mock_flags.py
>    <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/678/files#diff-1>
>    (88)
>    - *A* tests/mock_flags_test.py
>    <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/678/files#diff-2>
>    (74)
>
> Patch Links:
>
>    -
>    https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/678.patch
>    -
>    https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/678.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/678>.
>



-- 
Anthony F. Voellm (aka Tony)
Google Voice:  (650) 516-7382
<https://www.google.com/voice/b/0?pli=1#phones>
Blog: http://perfguy.blogspot.com/
Benchmark like a pro... PerfKitBenchmarker
<https://github.com/GoogleCloudPlatform/PerfKitBenchmarker>

File: tests/mock_flags_test.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    self.flags = mock_flags.MockFlags()
+
+  def testGetUnsetFlag(self):
+    self.assertIs(self.flags['test_flag'].present, False)
+
+  def testSetAndGetFlag(self):
+    self.flags.test_flag = 5
+    self.assertIs(self.flags.test_flag, 5)
+    self.assertIs(self.flags['test_flag'].present, True)
+    self.assertIs(self.flags['test_flag'].value, 5)
```
Comment (cmccoy) [inline tests/mock_flags_test.py]: Except for boolean and reference equality tests, I think this should be `assertEqual`

Comment (cmccoy) [inline tests/mock_flags_test.py]: (also below)

**Comment by cmccoy** [comment]
Nice. LGTM other than the comment about `assertIs` vs `assertEqual`.

File: tests/mock_flags_test.py (hunk 1 above)
Comment (skschneider) [inline tests/mock_flags_test.py]: Done.

**Pull request merged**